crates/solvers/src/domain/solver/baseline.rs

```diff
@@ -126,10 +126,30 @@ impl Inner {
             boundary::baseline::Solver::new(&self.weth, &self.base_tokens, &auction.liquidity);
 
         for (i, order) in auction.orders.into_iter().enumerate() {
-            let sell_token = auction.tokens.reference_price(&order.sell.token);
             let Some(user_order) = UserOrder::new(&order) else {
                 continue;
             };
+
+            let sell_token = match auction.tokens.reference_price(&order.sell.token) {
+                Some(price) => price,
+                None => {
+                    // Early return if the sell token is native token
+                    if user_order.get().sell.token == self.weth.0.into() {
+                        auction::Price(eth::Ether(eth::U256::exp10(18)))
+                    } else {
+                        // Estimate the price of the sell token in the native token
+                        let native_price_request = self.native_price_request(user_order);
+                        match boundary_solver.route(native_price_request, self.max_hops) {
+                            Some(route) => {
+                                let price = route.output().amount;
+                                auction::Price(eth::Ether(price))
+                            }
+                            None => continue,
```

Comment:
Instead of failing to give quote here, we could also default to `price=Inf` here instead, which would effectively lead to `fee = 0` which is the same imprecision we currently have (that I already described in the PR description ☝️ ) , but it would not require baseline to have [sell_token,native_token] liquidity to successfully return a quote.

wdyt?

Reply:
I think baseline is guaranteed to have [sell_token, native_token] liquidity by construction, so I'm not worried about that. 
I think in practice we aren't even using the fee estimate from quotes yet (cc @MartinquaXD). Solvers also specify a gas amount and we use a separate "native price request" to convert this into sell token. However, eventually we want to support solvers computing the sell token amount (not just gas amount), so preparing this change here without being too restrictive sounds good (that is not `continue` but return fee = 0)